source/ceylon/file/Resource.ceylon

```diff
@@ -34,9 +34,14 @@ shared interface ExistingResource
     shared formal void writeAttribute(Attribute attribute, 
                             Object attributeValue);
     
-}
-
+    "Request that this resource be deleted when the process terminates.
 
+     Invoking [[deleteOnExit()]] does not guarantee that the resource will
+     be deleted, and may have no effect on some [[System]]s or platforms.
+     For filesystem resources on the JVM, `deleteOnExit()` delegates to
+     `java.io.File.deleteOnExit()`."
+    shared formal void deleteOnExit();
```

Comment:
@jvasileff I think perhaps it would be more Ceylonic if this were an attribute, so you would write `file.deleteOnExit=true`, but I'm not sure if that's implementable.